include/mxnet/c_api_error.h

```diff
@@ -31,9 +31,35 @@
  * and finishes with API_END() or API_END_HANDLE_ERROR()
  * The finally clause contains procedure to cleanup states when an error happens.
  */
-#define MX_API_BEGIN() try { on_enter_api(__FUNCTION__);
-#define MX_API_END() } catch(dmlc::Error &_except_) { on_exit_api(); return MXAPIHandleException(_except_); } on_exit_api(); return 0;  // NOLINT(*)
-#define MX_API_END_HANDLE_ERROR(Finalize) } catch(dmlc::Error &_except_) { Finalize; on_exit_api(); return MXAPIHandleException(_except_); } on_exit_api(); return 0; // NOLINT(*)
+#define MX_API_BEGIN()                                                         \
+  try {                                                                        \
+    on_enter_api(__FUNCTION__);
+#define MX_API_END()                                                           \
+  }                                                                            \
+  catch (dmlc::Error &_except_) {                                              \
```

Comment:
Why do we need both? isn't dmlc::Error runtime_error? single catch should work.

Reply:
Good catch! I had kept this since i was planning on different error codes for frontend and this would differentiate between mxnet and non mxnet code. But since we are returning -1 for all currently I can remove it for now.